pkg/devfile/detect_test.go

```diff
@@ -91,6 +92,15 @@ func TestAnalyzeAndDetectDevfile(t *testing.T) {
 			registryURL: DevfileStageRegistryEndpoint,
 			wantErr:     true,
 		},
+		{
+			name:                "Component detected successfully with Port(s) detected",
```

Comment:
can we get coverage for a scenario where no ports are detected?

Reply:
The other existing tests cover those scenarios, as a result of alizer's relatively fragile port detection